### PR TITLE
feat(radar tracks msgs converter): implement dynamic status flag 'is_stationary' of the tracked object message, fix ego motion compensation

### PR DESCRIPTION
**Pull request by technolojin**
## Description

feat (radar_tracks_msgs_converter): The dynamic status flag (`is_stationary`) of the tracked object message is flagged based on the ground speed.

fix (radar_tracks_msgs_converter): In the radar ego motion compensation, the velocity compensation now considers the angular motion.

## Tests performed

* The dynamic status flag 
Simulations of sensing and perception modules were performed under a tier4 deployment.
Rviz plugin was tested.

https://github.com/autowarefoundation/autoware.universe/assets/42434141/eacbeeb3-89c7-48fb-b6d7-d78b228b6021

* Ego motion compensation fix
See videos in google drive(TIER IV internal).
    [before](https://drive.google.com/file/d/182694mXccPROxTz6sPLhZkTbJSE4P8G8/view?usp=drive_link)
    [after](https://drive.google.com/file/d/10cHnBRiBdu5qPQOCXxdM02XHoVvCFtel/view?usp=drive_link)

## Notes for reviewers
The original PR included rviz plugin modification. Now, it is separated to a new PR (https://github.com/autowarefoundation/autoware.universe/pull/5844). 

## Interface changes
Not applicable.

## Effects on system behavior
Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by YoshiRi — COMMENTED** [review]
LGTM.
But I have one comments and question.

## Comments

In current autoware universe development, we use yaml file to handle rosparam settings. So, we want to remove default value definition from cpp file and move settings from launch file to yaml file. See [sample PR](https://github.com/autowarefoundation/autoware.universe/pull/4756/files).

## Question

In the coordinate transformation and motion compensation part, this code is only assuming yaw angle. Does it assume radar sensor launched parallel to the ground or something?

**Comment by technolojin** [comment]
> ## Comments
> 
> In current autoware universe development, we use yaml file to handle rosparam settings. So, we want to remove default value definition from cpp file and move settings from launch file to yaml file. See [sample PR](https://github.com/autowarefoundation/autoware.universe/pull/4756/files).

The statement is acknowledged and the package will follow the policy in a separated PR. 

> ## Question
> 
> In the coordinate transformation and motion compensation part, this code is only assuming yaw angle. Does it assume radar sensor launched parallel to the ground or something?

That is true. Small angle of roll and pitch can be neglected. 

Additionally, the radar coordinate system is assumed to be upright. 
However, when the radar sensor can be installed upside down (180 deg roll) may due to vehicle integration limitation, then this conversion will be a false case.

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp [hunk 1]
```diff
@@ -307,6 +280,99 @@ TrackedObjects RadarTracksMsgsConverterNode::convertRadarTrackToTrackedObjects()
   return tracked_objects;
 }
 
+geometry_msgs::msg::Vector3 RadarTracksMsgsConverterNode::compensateVelocity(
```
Comment (scepter914) [inline perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp:283]: [IMO]
void compensateEgoMotion() is the same comment as void convertCovarianceMatrix()

File: perception/radar_tracks_msgs_converter/README.md [hunk 2]
```diff
@@ -20,10 +20,14 @@ This package convert from [radar_msgs/msg/RadarTracks](https://github.com/ros-pe
...
   - Default parameter is 20.0
-- `new_frame_id` (string): The header frame of output topic.
+- `new_frame_id` (string): The header frame of the output topic.
   - Default parameter is "base_link"
-- `use_twist_compensation` (bool): If the parameter is true, then the twist of output objects' topic is compensated by ego vehicle motion.
+- `use_twist_compensation` (bool): If the parameter is true, then the twist of the output objects' topic is compensated by ego vehicle motion.
   - Default parameter is "false"
+- `use_twist_yaw_compensation` (bool): If the parameter is true, then the ego motion compensation will also consider yaw motion of the ego vehicle.
+  - Default parameter is "false"
+- `static_obj_speed_threshold` (float): Specify the threshold for static object speed which determines the flag `is_stationary` [m/s].
```
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/README.md]: the parameter is fixed

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp [hunk 3]
```diff
@@ -209,91 +213,60 @@ TrackedObjects RadarTracksMsgsConverterNode::convertRadarTrackToTrackedObjects()
...
-      accel_cov[POSE_IDX::X_Y] = radar_accel_cov[RADAR_IDX::X_Y];
-      accel_cov[POSE_IDX::X_Z] = radar_accel_cov[RADAR_IDX::X_Z];
-      accel_cov[POSE_IDX::Y_X] = radar_accel_cov[RADAR_IDX::X_Y];
-      accel_cov[POSE_IDX::Y_Y] = radar_accel_cov[RADAR_IDX::Y_Y];
-      accel_cov[POSE_IDX::Y_Z] = radar_accel_cov[RADAR_IDX::Y_Z];
-      accel_cov[POSE_IDX::Z_X] = radar_accel_cov[RADAR_IDX::X_Z];
-      accel_cov[POSE_IDX::Z_Y] = radar_accel_cov[RADAR_IDX::Y_Z];
-      accel_cov[POSE_IDX::Z_Z] = radar_accel_cov[RADAR_IDX::Z_Z];
+    // determine static object
+    bool is_static_object = false;
```
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: the function definition is changed and tested.

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp [hunk 4]
```diff
@@ -209,91 +213,60 @@ TrackedObjects RadarTracksMsgsConverterNode::convertRadarTrackToTrackedObjects()
...
+    // heading is obtained from ground velocity
+    kinematics.pose_with_covariance.pose.orientation =
+      tier4_autoware_utils::createQuaternionFromYaw(yaw);
+    // longitudinal velocity is the length of the velocity vector
+    kinematics.twist_with_covariance.twist.linear.x = std::sqrt(
+      compensated_velocity.x * compensated_velocity.x +
+      compensated_velocity.y * compensated_velocity.y);
+    // lateral velocity is zero, use default value
+    // convert covariance matrices
+    convertCovarianceMatrix(radar_track, kinematics);
```
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: The conversion functions were defined.
This change causes copies of variables.

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp (hunk 1 above)
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: as same as above, the function returns value, and cases an additional copy operation.

File: perception/radar_tracks_msgs_converter/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/README.md]: fixed at [393d78d](https://github.com/autowarefoundation/autoware.universe/pull/5823/commits/393d78d65b7e029532dbe88645f0ee1e92d81967)

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: fixed at [a2bc40e](https://github.com/autowarefoundation/autoware.universe/pull/5823/commits/a2bc40e82ccd78025763a3fbe3344d17844839c9)

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp (hunk 1 above)
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: fixed at [7a35a7f](https://github.com/autowarefoundation/autoware.universe/pull/5823/commits/7a35a7f10b2ddff48fb259561afdb3483e40e7c5)

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: fixed at [5d0f010](https://github.com/autowarefoundation/autoware.universe/pull/5823/commits/5d0f010b602e8c6f460d9249b337db52dbdd5358)

**Review by scepter914 — APPROVED** [review]
LGTM

**Pull request merged**